NDTensors/src/UnallocatedArrays/src/unallocated_impl.jl

```diff
@@ -0,0 +1,27 @@
+for Typ in (:UnallocatedFill, :UnallocatedZeros)
+  ## Here are functions specifically defined for UnallocatedArrays
+  ## not implemented by FillArrays
+  ## TODO use set_parameters/get_parameters instead of alloctype and other 
+  ## type info functions.
+  ## TODO determine min number of functions needed to be forwarded
+  @eval begin
+    alloctype(A::$Typ) = alloctype(typeof(A))
+    alloctype(Atype::Type{<:$Typ}) = get_parameter(Atype, Position{4}())
+
+    Array(A::$Typ) = alloctype(typeof(A))
+
+    copy(A::$Typ) = A
+    ## TODO Implement vec
+    # Base.vec(Z::UnallocatedZeros) = typeof(Z)(length(Z))
+  end
+
+  ## TODO forwarding functions to fillarrays
+  ## convert doesn't work
+  for fun in (:size, :length, :sum, :getindex_value)
+    @eval FillArrays.$fun(A::$Typ) = $fun(parent(A))
+  end
```

Comment:
This seems a bit haphazard, for example `length` is usually derived from `size` for `AbstractArray` so generally doesn't need to be defined explicitly. Also I believe `FillArray` just needs a definition of `axes` from which `size` and `length` can be defined, though I'm not sure about that.

Also `sum` seems like a pretty random function for us to need to define.

Reply:
Again, I would look to https://github.com/JuliaArrays/FillArrays.jl/blob/7036a5005d2a96f79b67ffc9bfd4a1e07517c85e/src/FillArrays.jl#L281-L354 for guidance on the minimal interface we need to derive an `AbstractFill` subtype.

Reply:
I removed `@eval` and am going through the functions that are defined in the code block which you link to above, thanks!